airbyte_cdk/manifest_migrations/migrations_registry.py

```diff
@@ -0,0 +1,84 @@
+import importlib
+import inspect
+import pkgutil
+import re
+import sys
+from typing import List, Type
+
+import airbyte_cdk.manifest_migrations.migrations as migrations_pkg
+from airbyte_cdk.manifest_migrations.manifest_migration import (
+    ManifestMigration,
+)
+
+# Dynamically import all modules in the migrations package
+for _, module_name, is_pkg in pkgutil.iter_modules(migrations_pkg.__path__):
+    if not is_pkg:
+        importlib.import_module(f"{migrations_pkg.__name__}.{module_name}")
+
+
+def _migration_order_key(cls: object) -> int:
+    """
+    Determines the migration order key for a given migration class based on its module name.
+
+    The function expects the module name to end with a double underscore followed by an integer (e.g., '__0').
+    This integer is extracted and returned as the migration order key.
+
+    Args:
+        cls (object): The migration class whose module name encodes the migration order.
+
+    Returns:
+        int: The migration order extracted from the module name.
+
+    Raises:
+        ValueError: If the module name does not contain the expected order suffix.
+    """
+    # Extract the migration order from the module name, e.g., http_requester_url_base_to_url_v6_45_2__0
+    # The order is the integer after the double underscore at the end of the module name
+    module_name = cls.__module__.split(".")[-1]
+    match = re.search(r"__(\d+)$", module_name)
+    if match:
+        return int(match.group(1))
```

Comment:
I'm a bit confused by this one. I thought the naming convention was `<migration name>_v<semantic versioning>__<counter for this version>.py` but it seems to be `<migration name>_v<semantic versioning>__<counter for any version>.py`. Is that correct? 

This might be a nit (not sure) but it seems like it will be hard after a while for the dev to figure out what is this number because files aren't ordered by the migration order but they are usually ordered alphabetically in the file system which makes it hard to find where we're at in terms of migration

Reply:
I've reworked these patterns, no more versioning in the name of the class, and no more versioning + ordering in the name of the module, we now have the `registry.yaml` that collects all the migrations for the given manifest version. 

I've also updated the `README.md` to reflect this.